packages/beacon-node/src/network/gossip/encoding.ts

```diff
@@ -27,8 +27,7 @@ export function fastMsgIdFn(rpcMsg: RPC.IMessage): string {
 }
 
 export function msgIdToStrFn(msgId: Uint8Array): string {
-  // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
-  return Buffer.prototype.toString.call(msgId, "base64");
+  return toHex(msgId);
```

Comment:
are there any potential side-effects or considerations changing the return value from `base64` to `hex`?

Reply:
previously we choosed `base64` probably because of the memory efficiency, but it's an issue with Uint8Array anyway. Switching to `hex` helps us:
- fix the issue with Uint8Array by using `toHex` function
- have same message id format to other clients which make it easier to debug, for example Nimbus

I think the memory difference is no much

Reply:
sounds good to me, thanks for explaining 👍

Reply:
`toHex` util uses Buffer toString which is memory efficient too. Prefixing with "'0x" adds non-trivial unnecessary memory, but it's not that bad, and all hex strings are assumed to be prefixed so